Added Checklist and Known Issues section

edyuenyw opened the pull request:
Added initial changes for https://digital-engagement.atlassian.net/browse/SDPA-5804

edyuenyw committed:
Added Checklist and Known Issues section

anthony-malkoun commented on README.md:
```diff
@@ -3,8 +3,22 @@
 
 https://dpc-sdp.github.io/sdp-docs/
 
+
```
As discussed @edyuenyw `README.md` is about setting up the `dpc-sdp/sdp-docs` project, not about setting up the reference site. Everything under the `docs` directory in this project gets deployed (eventually) to https://dpc-sdp.github.io/sdp-docs/ so your changes should live under there. Thinking about this I'm thinking a new page under the `docs/working-with-sdp.md` for anything DPC/Cenitex specific and add to `docs/working-with-sdp.md` for other steps. Thoughts?

edyuenyw replied:
@anthony-malkoun I've made further changes and add more details to my commit. I also "reverted" the README.md commit which I still can see it in my log though there was not changed made to README.md anymore.
Can you please check and let me know if I need to change anything else?

anthony-malkoun reviewed (changes requested):
Hi @edyuenyw see my comments about getting these changes into the `docs` directory.

edyuenyw committed:
Revert "Added Checklist and Known Issues section"

This reverts commit b85bb89841d0d55bce12e12b03116a05251147bf.

edyuenyw committed:
SDPA-5804 Onboarding Changes
1. Create SSH key pair. Setting up GitHub and Lagoon to use the setup key.
2. Setup Lagoon CLI
3. Setup Pygmy
4. Setup Docker

edyuenyw committed:
Merge branch 'feature/SDPA-5804-update-onboarding-docs' of github.com:dpc-sdp/sdp-docs into feature/SDPA-5804-update-onboarding-docs

edyuenyw committed:
Merge branch 'master' of github.com:dpc-sdp/sdp-docs into feature/SDPA-5804-update-onboarding-docs

anthony-malkoun reviewed (changes requested):
Hi @edyuenyw can you update your build to use pygmy-go and document? I think there's value in creating a subpage on this for DPC/Cenitex with issues. You can create in this PR or create a new one if you prefer

edyuenyw committed:
Move setup instruction to Install dependencies section

edyuenyw committed:
SDPA-5804 Update onboarding document for a clean build

- Adds haproxy config .pygmy file reference

edyuenyw committed:
SDPA-5804 Update onboarding document for a clean build

- New file. Troubleshooting guide for working-with-sdp page

edyuenyw committed:
Added more detailed steps for Cenitex issue

anthony-malkoun commented on mkdocs.yml:
```diff
@@ -97,6 +97,7 @@ markdown_extensions:
 nav:
   - About: index.md
   - Working with SDP: working-with-sdp.md
+  - Troubleshooting: troubleshooting.md
```
I think this should be a subpage of the `working-with-sdp.md` page. Just indent (see Bay: below)

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+    You are viewing a version of the documentation that is currently under 
+    active development.
+
+ This page provides some workaround for issues that you may encounter during a local build.
```
*workarounds

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+ This page provides some workaround for issues that you may encounter during a local build.
+
+## Open Workspace | Cenitex | Citrix
+Unable to install software (e.g Docker Desktop) or run commands which requires elevated permissions.
```
*require

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+
+1. Check if you Open Workspace is in your machine. This will allow software installation that requires elevated permissions.
+2. If you believe to have Open Workspace installed but still have issue with installations, logoff from Citrix Gateway (Windows Taskbar Icon) and reconnect through Go Connect (Window search). An IE browser will open and ask for your DPC login credentials. Process might take some time to complete for others and for some, browser seemed to be hanging. In that case try restarting your computer and r ...
+3. Check again to see if you have the elevated permissions or not. You might have restart your computer several times to take effect.
```
*have to restart

anthony-malkoun replied:
*several times for it it to take effect.

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+2. If you believe to have Open Workspace installed but still have issue with installations, logoff from Citrix Gateway (Windows Taskbar Icon) and reconnect through Go Connect (Window search). An IE browser will open and ask for your DPC login credentials. Process might take some time to complete for others and for some, browser seemed to be hanging. In that case try restarting your computer and r ...
+3. Check again to see if you have the elevated permissions or not. You might have restart your computer several times to take effect.
+
+## SSH Key
```
Delete if no issues.

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+*(add ssh key issue if any)*
+
+## Lagoon | Lagoon CLI
+*(Remote work)* Unable to resolve host or cannot descrypt encoded SSH
```
*decrypt

anthony-malkoun commented on docs/working-with-sdp.md:
```diff
@@ -26,12 +26,71 @@ with the DPC team there are a few tasks to complete.
-Once you have access to the [Lagoon Dashboard](https://dashboard.amazeeio.cloud/projects/) make sure to add your
+### Create SSH key pair
+
+- [Generate new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) will guide you on how to generate, test and also add your key to your GitHub [*(Add a new SSH key)*](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
```
Did the instruction specify a different key type then `id_rsa`? If so don't we need to make a note to only use `id_rsa` and not the suggested one?

anthony-malkoun commented on docs/working-with-sdp.md:
````diff
@@ -26,12 +26,71 @@ with the DPC team there are a few tasks to complete.
+- Download and Install Docker Desktop (https://www.docker.com/get-started).
+- After installation, run:
+
+    ```
````
Please explain the differences between `ahoy build` (destroys all containers and rebuilds from scratch) and `ahoy up` which just starts existing containers.

anthony-malkoun commented on docs/working-with-sdp.md:
````diff
@@ -26,12 +26,71 @@ with the DPC team there are a few tasks to complete.
+    pygmy-go up
+    ```
+
+- Create a new .pygmy.yml file so haproxy can be configured to another port other than 80
````
This should go into the troubleshooting document IMO as it's a Cenitex only issue (as it hogs port 80)

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+## Open Workspace | Cenitex | Citrix
+Unable to install software (e.g Docker Desktop) or run commands which requires elevated permissions.
+
+1. Check if you Open Workspace is in your machine. This will allow software installation that requires elevated permissions.
```
*Check if Open Workplace is installed on your machine.

Also, a screen shot of the PCInfo app would help here. Ping me if you need help with this.

anthony-malkoun commented on docs/troubleshooting.md:
```diff
@@ -0,0 +1,53 @@
+Unable to install software (e.g Docker Desktop) or run commands which requires elevated permissions.
+
+1. Check if you Open Workspace is in your machine. This will allow software installation that requires elevated permissions.
+2. If you believe to have Open Workspace installed but still have issue with installations, logoff from Citrix Gateway (Windows Taskbar Icon) and reconnect through Go Connect (Window search). An IE browser will open and ask for your DPC login credentials. Process might take some time to complete for others and for some, browser seemed to be hanging. In that case try restarting your computer and r ...
```
```suggestion
2. If you believe Open Workplace is installed but are still having issues with installations, logoff from Citrix Gateway (Windows Taskbar Icon) and reconnect through Go Connect (Window search). An IE browser will open and ask for your DPC login credentials. Process might take some time to complete or your browser might appear to hang. In that case try restarting your computer and retry Go Connect.
```

anthony-malkoun reviewed (changes requested):
Hi @edyuenyw most of the changes are small grammar/spelling issues. There are a few bigger changes I'd like to see. Let me know if you need more detail.

edyuenyw committed:
SDPA 5804 Update Onboarding Docs

- Updates from feedbacks
- Changes Working With SDP Navigation Menu to have sub menus

anthony-malkoun reviewed (approved):
Thanks for making the requested changes @edyuenyw. Please feel free to merge this in when you get a chance.
